### PR TITLE
add check if .html file exist in source of zip file, otherwise return Error

### PR DESCRIPTION
**Pull request by B-Naoufal**
… an error WebCreator-1005

### CODE REVIEW

File: int/api/websites/webcreator.go [hunk 1]
```diff
@@ -55,6 +82,18 @@ func prepareForWebsiteHandler(params operations.WebsiteCreatorPrepareParams, app
...
+	// Read all the files from zip archive
+	for _, zipFile := range zipReader.File {
+		if err != nil {
+			return nil
+		}
+		list_of_files = append(list_of_files, zipFile.Name)
+	}
+	// check if zip archive exist
+	if !contains(list_of_files, "index.html") {
+		return createInternalServerError(errorCodeWebCreatorHtmlNotInSource, err.Error())
```
Comment (Thykof) [inline int/api/websites/webcreator.go:95]:
```suggestion
		return createInternalServerError(errorCodeWebCreatorHTMLNotInSource, err.Error())
```

**Review by Thykof — CHANGES_REQUESTED** [review]
There are some linter issues

**Comment by B-Naoufal** [comment]
@Thykof just committed the suggestion modifications

**Review by Thykof — CHANGES_REQUESTED** [review]
there is a changed file when I run `go generate` I think we should commit it

**Comment by B-Naoufal** [comment]
> 

Done missed go generate for embedded spec

**Review by Thykof — CHANGES_REQUESTED** [review]

**Review by Thykof — APPROVED** [review]

**Review by thomas-senechal — APPROVED** [review]

File: int/api/websites/webcreator.go [hunk 2]
```diff
@@ -28,6 +30,25 @@ func CreatePrepareForWebsiteHandler(
 	}
 }
 
+func listArchiveContent(zipReader *zip.Reader) []string {
```
Comment (gregLibert) [inline int/api/websites/webcreator.go:33]: You function name is misleading, you should name it following what the code is doing.

My proposal:
```suggestion
func listFileName(zipReader *zip.Reader) []string {
```

File: int/api/websites/webcreator.go [hunk 3]
```diff
@@ -28,6 +30,25 @@ func CreatePrepareForWebsiteHandler(
...
+
+func contains(elems []string, v string) bool {
+	for _, s := range elems {
+		if v == s {
+			return true
+		}
+	}
+
+	return false
+}
```
Comment (gregLibert) [inline int/api/websites/webcreator.go:50]: The best code is the one you're not writing : https://pkg.go.dev/golang.org/x/exp/slices#Index

File: int/api/websites/webcreator.go [hunk 4]
```diff
@@ -61,8 +82,11 @@ func prepareForWebsiteHandler(params operations.WebsiteCreatorPrepareParams, app
 		return createInternalServerError(errorCodeWebCreatorArchiveSize, errorCodeWebCreatorArchiveSize)
 	}
 
-	if !checkContentType(archive, "application/zip") {
-		return createInternalServerError(errorCodeWebCreatorFileType, errorCodeWebCreatorFileType)
+	zipReader, _ := zip.NewReader(bytes.NewReader(archive), int64(len(archive)))
+	FilesOfArchive := listArchiveContent(zipReader)
+
+	if !contains(FilesOfArchive, "index.html") {
```
Comment (gregLibert) [inline int/api/websites/webcreator.go:88]: should be 
```suggestion
	if slice.Index(FilesOfArchive, "index.html") == -1 {
```

**Review by gregLibert — CHANGES_REQUESTED** [review]
:no_good: 
You should use the native function instead of creating your own find in array.

The small suggestion for code clarity is more nitpicking (I would not have shared it without another comment).

**Review by gregLibert — APPROVED** [review]
👏 good job!

**Pull request merged**